vassal-app/src/main/java/VASSAL/counters/UsePrototype.java

```diff
@@ -323,6 +336,88 @@ public String getState() {
     public String getType() {
       return ID + nameConfig.getValueString();
     }
+
+    private String getValueString() {
+      return nameConfig.getValueString();
+    }
+
+    private static class Validator extends JLabel {
+
+      protected static final int INVALID = 0;
+      protected static final int VALID = 1;
+      protected static final int UNKNOWN = 2;
+
+      protected final Icon tick;
+      protected final Icon cross;
+      protected final ImageIcon none;
+      protected int status = UNKNOWN;
+      protected boolean validating = false;
+      protected boolean dirty = false;
+      protected final ValidationThread validationThread = new ValidationThread();
+      protected final Editor editor;
+
+      private static final long serialVersionUID = 1L;
+
+      public Validator(Editor editor) {
+        this.editor = editor;
+        cross = IconFactory.getIcon("no", IconFamily.XSMALL);  //NON-NLS
+        tick = IconFactory.getIcon("yes", IconFamily.XSMALL); //NON-NLS
+
+        final BufferedImage image = new BufferedImage(cross.getIconWidth(), cross.getIconHeight(), BufferedImage.TYPE_INT_ARGB);
+        none = new ImageIcon(image);
+
+        setStatus(UNKNOWN);
+      }
+
+      public void setStatus(int status) {
+        if (status == VALID) {
+          setIcon(tick);
+        }
+        else if (status == INVALID) {
+          setIcon(cross);
+        }
+        else {
+          setIcon(none);
+        }
+        this.status = status;
+        repaint();
+      }
+
+      /*
+       *  Run the validation in a separate thread. If the expression is updated
+       *  while validating, then revalidate.
+       */
+      @Override
+      public void validate() {
+        if (validating) {
+          dirty = true;
+        }
+        else {
+          validating = true;
+          setStatus(UNKNOWN);
+          SwingUtilities.invokeLater(validationThread);
+        }
+      }
+
+      class ValidationThread implements Runnable {
```

Comment:
Should be private?

Reply:
Actually, everything in there should be private. Updated